### PR TITLE
Update go doc

### PR DESCRIPTION
**Pull request by AlexAPB**
This updates the yarpc godoc with currently available features.

### CODE REVIEW

**Comment by AlexAPB** [comment]
I'm not going to pretend like I know all of what's going on in yarpc-go right now, but it seems as if this godoc was stale. Let me know what's missing or the best way to write this out (@kriskowal).

File: doc.go [hunk 1]
```diff
@@ -30,12 +30,12 @@
...
 // YARPC breaks RPC into interchangeable encodings, transports, and peer
 // choosers.
-// YARPC for Go provides reference implementations for HTTP/1.1 and TChannel
-// transports, and also raw, JSON, and Thrift encodings.
-// YARPC for Go provides experimental implementations for a Redis transport and a
-// round robin peer chooser.
-// YARPC for Go plans to provide a Protobuf 3 encoding, a gRPC transport, and a
-// load balancer that uses a least-pending-requests strategy.
+// YARPC for Go provides reference implementations for HTTP/1.1, TChannel and gRPC
+// transports, and also raw, JSON, Thrift and Protobuf 3 encodings.
```
Comment (peter-edge) [inline doc.go]: s/Protobuf 3/Protobuf/

**Comment by kriskowal** [comment]
@akshayjshah I think you’re noticing a conceit of mine, that I always start a new line for each sentence. My reasoning is that it reduces diff churn, where paragraph wrapping doesn't cascade across sentences, and I don’t have to reveal my bad opinions about spaces after periods. My intention here was a single paragraph.

File: doc.go [hunk 2]
```diff
@@ -30,12 +30,12 @@
...
 // YARPC breaks RPC into interchangeable encodings, transports, and peer
 // choosers.
-// YARPC for Go provides reference implementations for HTTP/1.1 and TChannel
-// transports, and also raw, JSON, and Thrift encodings.
-// YARPC for Go provides experimental implementations for a Redis transport and a
-// round robin peer chooser.
-// YARPC for Go plans to provide a Protobuf 3 encoding, a gRPC transport, and a
-// load balancer that uses a least-pending-requests strategy.
+// YARPC for Go provides reference implementations for HTTP/1.1, TChannel and gRPC
+// transports, and also raw, JSON, Thrift and Protobuf encodings.
```
Comment (kriskowal) [inline doc.go]: This is a Chicago manual (Microsoft manual) shop. Comma before conjunction.

Comment (AlexAPB) [reply doc.go]: I will reluctantly add this comma :)

**Comment by akshayjshah** [comment]
@kriskowal ahhh, that's smart. I may adopt that idea myself :)

**Pull request merged**